src/backends/ncnn/ncnnlib.cc

```diff
@@ -281,6 +281,21 @@ namespace dd
 
     if (bbox == true)
       {
+        std::string uri = inputc._ids.at(0);
+        auto bit = inputc._imgs_size.find(uri);
+        int rows = 1;
+        int cols = 1;
+        if (bit != inputc._imgs_size.end())
+          {
+            // original image size
+            rows = (*bit).second.first;
+            cols = (*bit).second.second;
+          }
+        else
+          {
+            this->_logger->error("couldn't find original image size for {}",
+                                 uri);
```

Comment:
Throw an error maybe? If rows = cols = 1, then we divide by 0 later in the code, leading to probably NaN bbox, which is not easy to debug.

Reply:
Good catch, I've updated the PR with an internal exception, and added another commit for having the same exact behavior with the caffe backend.